packages/loader/container-loader/src/containerContext.ts

```diff
@@ -91,6 +91,10 @@ export class ContainerContext implements IContainerContext {
 
     public readonly taggedLogger: ITelemetryLogger;
 
+    public setReadOnly(): void {
```

Comment:
Couple thoughts. First, it helps to use same verbs / names across layers, if possible.
So forceReadonly() might be advised here.

That said, usage of forceReadonly() means that host can easily overwrite it by calling forceReadonly(false). I do not think that's what you want.

And most importantly - we should be careful to add more APIs to layers as it complicates ability to build alternative implementations of those layers. Can we get away with existing flow? We already have a flow of throwing critical errors that close container, and host already knows how to deal with them (errorType). And in fact, host would need to know about this case to show some reasonable error dialog to user - just setting read-only state is not enough as it has no info on what it happened.

The minus of going with critical errors is that client stops receiving remote changes. But I think that's fine